src/config.js

```diff
@@ -0,0 +1,1602 @@
+        return result.then(endTransaction(resolve)).catch(endTransaction(reject))
+      })
+    } catch (error) {
+      this.endTransaction()
+      return Promise.reject(error)
+    }
+  }
+
+  beginTransaction () {
+    this.transactDepth++
+  }
+
+  endTransaction () {
+    this.transactDepth--
+    this.emitChangeEvent()
+  }
+
+  pushAtKeyPath (keyPath, value) {
+    let left
+    const arrayValue = (left = this.get(keyPath)) != null ? left : []
+    const result = arrayValue.push(value)
+    this.set(keyPath, arrayValue)
+    return result
+  }
+
+  unshiftAtKeyPath (keyPath, value) {
+    let left
+    const arrayValue = (left = this.get(keyPath)) != null ? left : []
+    const result = arrayValue.unshift(value)
+    this.set(keyPath, arrayValue)
+    return result
+  }
+
+  removeAtKeyPath (keyPath, value) {
+    let left
+    const arrayValue = (left = this.get(keyPath)) != null ? left : []
+    const result = _.remove(arrayValue, value)
+    this.set(keyPath, arrayValue)
+    return result
+  }
+
+  setSchema (keyPath, schema) {
+    if (!isPlainObject(schema)) {
+      throw new Error(`Error loading schema for ${keyPath}: schemas can only be objects!`)
+    }
+
+    if (schema.type == null) {
+      throw new Error(`Error loading schema for ${keyPath}: schema objects must have a type attribute`)
+    }
+
+    let rootSchema = this.schema
+    if (keyPath) {
+      for (let key of Array.from(splitKeyPath(keyPath))) {
+        rootSchema.type = 'object'
+        if (rootSchema.properties == null) { rootSchema.properties = {} }
+        const { properties } = rootSchema
+        if (properties[key] == null) { properties[key] = {} }
+        rootSchema = properties[key]
+      }
+    }
+
+    Object.assign(rootSchema, schema)
+    return this.transact(() => {
+      this.setDefaults(keyPath, this.extractDefaultsFromSchema(schema))
+      this.setScopedDefaultsFromSchema(keyPath, schema)
+      return this.resetSettingsForSchemaChange()
+    })
+  }
+
+  load () {
+    this.initializeConfigDirectory()
+    this.loadUserConfig()
+    return this.observeUserConfig()
+  }
+
+  /*
+  Section: Private methods managing the user's config file
+  */
+  initializeConfigDirectory (done) {
+    if (fs.existsSync(this.configDirPath) || this.shouldNotAccessFileSystem()) { return }
+
+    fs.makeTreeSync(this.configDirPath)
+
+    const queue = async.queue(({sourcePath, destinationPath}, callback) => fs.copy(sourcePath, destinationPath, callback))
+    queue.drain = done
+
+    const templateConfigDirPath = fs.resolve(this.resourcePath, 'dot-atom')
+    const onConfigDirFile = sourcePath => {
+      const relativePath = sourcePath.substring(templateConfigDirPath.length + 1)
+      const destinationPath = path.join(this.configDirPath, relativePath)
+      return queue.push({sourcePath, destinationPath})
+    }
+    return fs.traverseTree(templateConfigDirPath, onConfigDirFile, path => true, () => {})
+  }
+
+  loadUserConfig () {
+    let error
```

Comment:
are the errors here actually being used outside their catch blocks? i think this can be removed and you can just use `error` instead of (`error1` and `error2`) in the catch blocks below.